code/game/objects/items/storage/bags.dm

```diff
@@ -574,27 +574,69 @@
 		new /obj/item/ammo_casing/caseless/harpoon(src)
 
 /obj/item/storage/bag/rebar_quiver
-	name = "Rebar Storage Quiver"
-	icon = 'icons/obj/weapons/guns/bows/quivers.dmi'
+	name = "rebar quiver"
+	icon = 'icons/obj/weapons/bows/quivers.dmi'
 	icon_state = "rebar_quiver"
 	worn_icon_state = "rebar_quiver"
 	inhand_icon_state = "rebar_quiver"
 	desc = "A oxygen tank cut in half, used for holding sharpened rods for the rebar crossbow."
-	slot_flags = ITEM_SLOT_BACK|ITEM_SLOT_SUITSTORE
+	slot_flags = ITEM_SLOT_BACK|ITEM_SLOT_SUITSTORE|ITEM_SLOT_NECK
 	resistance_flags = FLAMMABLE
+	storage_type = /datum/storage/bag/rebar_quiver
+
+/obj/item/storage/bag/rebar_quiver/syndicate
+	icon_state = "syndie_quiver_0"
+	worn_icon_state = "syndie_quiver_0"
+	inhand_icon_state = "holyquiver"
+	desc = "A specialized quiver meant to hold any kind of bolts intended for use with the rebar crossbow. \
+		Clearly a better design than a cut up oxygen tank..."
+	slot_flags = ITEM_SLOT_NECK
+	w_class = WEIGHT_CLASS_NORMAL
+	resistance_flags = LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+	actions_types = list(/datum/action/item_action/reload_rebar)
+	action_slots = ALL
+	storage_type = /datum/storage/bag/rebar_quiver/syndicate
 
-/obj/item/storage/bag/rebar_quiver/Initialize(mapload)
+/obj/item/storage/bag/rebar_quiver/syndicate/Initialize(mapload)
 	. = ..()
-	atom_storage.max_specific_storage = WEIGHT_CLASS_TINY
-	atom_storage.max_slots = 10
-	atom_storage.max_total_storage = 15
-	atom_storage.set_holdable(list(
-		/obj/item/ammo_casing/rebar,
-		/obj/item/ammo_casing/rebar/syndie,
-		/obj/item/ammo_casing/rebar/healium,
-		/obj/item/ammo_casing/rebar/hydrogen,
-		/obj/item/ammo_casing/rebar/zaukerite,
-		/obj/item/ammo_casing/rebar/paperball,
-		))
+	update_appearance(UPDATE_OVERLAYS)
+
+/obj/item/storage/bag/rebar_quiver/syndicate/PopulateContents()
+	for(var/to_fill in 1 to 20)
+		new /obj/item/ammo_casing/rebar/syndie(src)
+
+/obj/item/storage/bag/rebar_quiver/syndicate/update_icon_state()
+	. = ..()
+	switch(contents.len)
+		if(0)
+			icon_state = "syndie_quiver_0"
+		if(1 to 7)
+			icon_state = "syndie_quiver_1"
+		if(8 to 13)
+			icon_state = "syndie_quiver_2"
+		if(14 to 20)
+			icon_state = "syndie_quiver_3"
```

Comment:
```suggestion
			icon_state = "syndie_quiver_0"
		if(1 to 7)
			icon_state = "syndie_quiver_1"
		if(8 to 13)
			icon_state = "syndie_quiver_2"
		if(14 to 20)
			icon_state = "syndie_quiver_3"
```
wouldn't it be better to have `base_icon_state = "syndie_quiver"` on the typepath, and then have these like `icon_state = "[base_icon_state]_0"`